pkg/block/block.go

```diff
@@ -42,9 +42,28 @@ const (
 	DebugMetas = "debug/metas"
 )
 
+// downloadDirFn is a function type that a function which downloads a directory from remote object
+// storage must implement. Used for mocking.
+type downloadDirFn func(context.Context, log.Logger, objstore.BucketReader, string, string, []string) error
```

Comment:
This is bit confusing. Can we just have deleteDir function directly used and mock object storage?